genesyscloud/architect_flow/data_source_genesyscloud_flow.go

```diff
@@ -0,0 +1,43 @@
+package architect_flow
+
+import (
+	"context"
+	"fmt"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceFlowRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	sdkConfig := m.(*provider.ProviderMeta).ClientConfig
+	p := getArchitectFlowProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+
+	// Query flow by name. Retry in case search has not yet indexed the flow.
+	return util.WithRetries(ctx, 5*time.Second, func() *retry.RetryError {
+		const pageSize = 100
+		for pageNum := 1; ; pageNum++ {
```

Comment:
We tend to use function, getXByName, in the datasource

Reply:
Not going to address right now